### PR TITLE
Update certs

### PR DESCRIPTION
**Pull request by jleclanche**
## Todos
- [x] MIT compatible
- [x] Tests
- [x] Documentation
- [x] Updated CHANGES.rst

### CODE REVIEW

**Comment by jleclanche** [comment]
This supercedes #130 and fixes the tests (or at least should)

File: tests/test_default_ssl.py [hunk 1]
```diff
@@ -1,28 +1,10 @@
...
-vLihgYukgYgwgYUxCzAJBgNVBAYTAlVTMRMwEQYDVQQIEwpXYXNoaW5ndG9uMRAw
-DgYDVQQHEwdTZWF0dGxlMRMwEQYDVQQKEwpBbWF6b24uY29tMQswCQYDVQQLEwJD
-TTEtMCsGCSqGSIb3DQEJARYeY29va2llLW1vbnN0ZXItY29yZUBhbWF6b24uY29t
-ggkAs8c8NyOT3v0wDAYDVR0TBAUwAwEB/zANBgkqhkiG9w0BAQUFAAOBgQCYZSRQ
-zJNHXyKACrqMB5j1baUGf5NA0cZ/8s5iWeC9Gkwi7cXyiq9OrBaUtJBzAJTzfWbH
-dfVaBL5FWuQsbkJWHe0mV+l4Kzl5bh/FSDSkhYR1duYRmdCXckQk6mAF6xG+1mpn
-8YlJmbEhkDmBgJ8C8p0LCMNaO2xFLlNU0O+0ng==
------END CERTIFICATE-----
-"""
+CERT_PATH = resource_filename("sqlalchemy_redshift", "redshift-ca-bundle.crt")
```
Comment (graingert) [inline tests/test_default_ssl.py]: resoure_string

Comment (jleclanche) [reply tests/test_default_ssl.py]: Fixed

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
   matrix:
-  - TOX_ENV=py26
-  - TOX_ENV=py27
-  - TOX_ENV=py33
-  - TOX_ENV=py34
-  - TOX_ENV=py35
-  - TOX_ENV=lint
+    - TOXENV=py27
+    - TOXENV=py34
+    - TOXENV=py35
```
Comment (graingert) [inline .travis.yml]: just remove this line

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
+    - secure: g/ta2yudfA55ruxd+GHxYQUd39OM/hoEhNp0KgPm3QINaXwaB6C9eQ31qOi/i9OiSjmVwISE+yUaPtQP2jtbYnkS/SHiRNvzhfA+mPrT0n0hoVSNIlxl94E8hoPBZr2jlYVQ6dIuT98mIRYJu/wiGHDQCuar6znpiKFB1/gyUszvZxAxHJIe5EcjuXWFnre91b96tNkZRKJZv+Jst0CkknVcKAZ4kmC/o5Z1hr5VaMekRV5SKf+vMEfVc+W9VktqDBgKwaFIwWqSgQFk5LOnPr/EZq8mrqPpa+LFaLiQpDGDAdaVEM0/A3dzx3YnYoBFNr5hmCtoCLWccxC+diaw3kZuTX6olxn8cbRCVf8lu4JuJKFU5CgTLS7NBH3Rpk4iRE7C9v4W+QKjw3CJxLCMoMqQCbYx6GjzjZYbZBA6jvaS6ZFe41N6egA0SpNvKtfYU9TR7lNVwGehRIY9kycOEX5ZQInkqILxz5YHrkMNfICdLY+rYFBDvgNAOZxYR1tLaOG9jYuUbBOLVRZz78wkPrXynquay6HZn7AxBxOdnvGQAkINvo40Lx/MWdZ88o3f/th838cpPCxBRQZ5Kn3RkHNhGHwUJadsbaOLnKrj3ISaTOvq0brA1mxJFyJwoVrn+rFs80CWRonQ5lQeaOHf0p1flH7j+TFhl/INkvXsf7Y=  # PGPASSWORD
+
+matrix:
+  fast_finish: true
+  include:
+    - python: "3.5"
+      env: TOXENV=py35
+  exclude:
```
Comment (graingert) [inline .travis.yml]: no need to exclude this

File: .travis.yml [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+    - secure: g/ta2yudfA55ruxd+GHxYQUd39OM/hoEhNp0KgPm3QINaXwaB6C9eQ31qOi/i9OiSjmVwISE+yUaPtQP2jtbYnkS/SHiRNvzhfA+mPrT0n0hoVSNIlxl94E8hoPBZr2jlYVQ6dIuT98mIRYJu/wiGHDQCuar6znpiKFB1/gyUszvZxAxHJIe5EcjuXWFnre91b96tNkZRKJZv+Jst0CkknVcKAZ4kmC/o5Z1hr5VaMekRV5SKf+vMEfVc+W9VktqDBgKwaFIwWqSgQFk5LOnPr/EZq8mrqPpa+LFaLiQpDGDAdaVEM0/A3dzx3YnYoBFNr5hmCtoCLWccxC+diaw3kZuTX6olxn8cbRCVf8lu4JuJKFU5CgTLS7NBH3Rpk4iRE7C9v4W+QKjw3CJxLCMoMqQCbYx6GjzjZYbZBA6jvaS6ZFe41N6egA0SpNvKtfYU9TR7lNVwGehRIY9kycOEX5ZQInkqILxz5YHrkMNfICdLY+rYFBDvgNAOZxYR1tLaOG9jYuUbBOLVRZz78wkPrXynquay6HZn7AxBxOdnvGQAkINvo40Lx/MWdZ88o3f/th838cpPCxBRQZ5Kn3RkHNhGHwUJadsbaOLnKrj3ISaTOvq0brA1mxJFyJwoVrn+rFs80CWRonQ5lQeaOHf0p1flH7j+TFhl/INkvXsf7Y=  # PGPASSWORD
+
+matrix:
+  fast_finish: true
```
Comment (graingert) [inline .travis.yml]: ??

File: .travis.yml (hunk 3 above)
Comment (jleclanche) [reply .travis.yml]: If you don't exclude it, you'll end up with failing builds.

File: .travis.yml (hunk 4 above)
Comment (jleclanche) [reply .travis.yml]: https://blog.travis-ci.com/2013-11-27-fast-finishing-builds/

File: .travis.yml (hunk 2 above)
Comment (jleclanche) [reply .travis.yml]: Remove what?

Comment (graingert) [reply .travis.yml]: This line

File: .travis.yml (hunk 3 above)
Comment (graingert) [reply .travis.yml]: Try it after removing the line I suggested

File: .travis.yml (hunk 2 above)
Comment (jleclanche) [reply .travis.yml]: I don't understand. What do you want me to remove and why? Do you want to stop testing on py35?

Comment (graingert) [reply .travis.yml]: Please delete this line. You don't need to add it here then exclude it again on line 21

Comment (jleclanche) [reply .travis.yml]: Please be more specific. What line? If I delete `TOXENV=py35` then we're no longer testing on Python 3.5. I'm fine with that personally but I don't think it's what you want.

Comment (graingert) [reply .travis.yml]: That's not true. Try deleting this line and the exclude block and see what happens

File: .travis.yml (hunk 4 above)
Comment (graingert) [reply .travis.yml]: We don't have any allowed failures so I'm pretty sure it's redundant

Comment (jleclanche) [reply .travis.yml]: That's fair. I copied this from my usual travis template; I can remove it but it's good practice to leave it in imo.

**Comment by jleclanche** [comment]
@graingert I've pushed the change you requested.

**Comment by graingert** [comment]
The commit message is wrong, note how the tests still run on 3.5: https://travis-ci.org/sqlalchemy-redshift/sqlalchemy-redshift/jobs/279172158

**Comment by jleclanche** [comment]
@graingert all good now? I'd really like to see this merged and a new release before october so that we're ready to transition :)

**Comment by graingert** [comment]
Yeah an October release is totally doable

**Comment by jleclanche** [comment]
@graingert So good to merge then?

**Comment by graingert** [comment]
@jleclanche can you rebase please?

File: setup.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
     license="MIT",
     url='https://github.com/sqlalchemy-redshift/sqlalchemy-redshift',
     packages=['sqlalchemy_redshift', 'redshift_sqlalchemy'],
-    package_data={'sqlalchemy_redshift': ['redshift-ssl-ca-cert.pem']},
+    package_data={'sqlalchemy_redshift': ['redshift-ca-bundle.crt']},
```
Comment (graingert) [inline setup.py]: probably a breaking change, flagging for 0.7.x

File: tests/test_default_ssl.py [hunk 6]
```diff
@@ -1,28 +1,8 @@
...
-vLihgYukgYgwgYUxCzAJBgNVBAYTAlVTMRMwEQYDVQQIEwpXYXNoaW5ndG9uMRAw
-DgYDVQQHEwdTZWF0dGxlMRMwEQYDVQQKEwpBbWF6b24uY29tMQswCQYDVQQLEwJD
-TTEtMCsGCSqGSIb3DQEJARYeY29va2llLW1vbnN0ZXItY29yZUBhbWF6b24uY29t
-ggkAs8c8NyOT3v0wDAYDVR0TBAUwAwEB/zANBgkqhkiG9w0BAQUFAAOBgQCYZSRQ
-zJNHXyKACrqMB5j1baUGf5NA0cZ/8s5iWeC9Gkwi7cXyiq9OrBaUtJBzAJTzfWbH
-dfVaBL5FWuQsbkJWHe0mV+l4Kzl5bh/FSDSkhYR1duYRmdCXckQk6mAF6xG+1mpn
-8YlJmbEhkDmBgJ8C8p0LCMNaO2xFLlNU0O+0ng==
------END CERTIFICATE-----
-"""
+CERT = resource_string("sqlalchemy_redshift", "redshift-ca-bundle.crt")
```
[the remark this answers is not in the document]
Comment (graingert) [reply tests/test_default_ssl.py]: or maybe just make the test:

CERT_PATH = resource_filename("sqlalchemy_redshift", "redshift-ca-bundle.crt")

...

assert cparams.pop('sslrootcert') == CERT_PATH

**Comment by jleclanche** [comment]
Rebased.

What do you mean by "the test is recursive"?

File: tests/test_default_ssl.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (graingert) [reply tests/test_default_ssl.py]: So the test used to say:

> The code should make sure these bytes end up in the sslrootcert parameter file.


Now it says:


> The code should make sure the bytes defined by the code end up in the sslrootcert parameter file.

**Comment by graingert** [comment]
> What do you mean by "the test is recursive"?

@jleclanche see thread.

**Comment by jleclanche** [comment]
I saw your suggestion but I still don't see what you're achieving/avoiding. The tests pass for me and travis?

File: tests/test_default_ssl.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (graingert) [reply tests/test_default_ssl.py]: I'm trying to say that the tests should avoid using values imported from the code as part of their expectations.

**Comment by graingert** [comment]
> I saw your suggestion but I still don't see what you're achieving/avoiding. The tests pass for me and travis?

@jleclanche see thread.

File: tests/test_default_ssl.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jleclanche) [reply tests/test_default_ssl.py]: Right, I see what you mean. I don't really think a copy/paste test is the way to go though (that's more error-prone and not really checking anything). I think as-is the test at least checks that the cert exists and is available in the package data.

Aren't the live redshift tests ensuring that ssl connectivity works?

**Comment by graingert** [comment]
Yup. I think the CERT_PATH test would be the best

**Comment by jleclanche** [comment]
@graingert Updated test as per your suggestion.

**Comment by graingert** [comment]
@jleclanche nice stuff! Any chance you want to help out with https://github.com/sqlalchemy-redshift/sqlalchemy-redshift/issues/131 ?

**Comment by jleclanche** [comment]
@graingert You already landed my travis fixes which should get the build green again. I'm not too sure why the latest builds are failing but it looks unrelated to the repository (looks like comms failure between travis/github...)

**Comment by jcalero** [comment]
Is this due to be released before October 23 when AWS will stop accepting the old certificate?

**Pull request merged**